clang/runtime/dpct-rt/include/dpl_extras/algorithm.h.inc

```diff
@@ -1452,6 +1452,47 @@ inline void segmented_sort_pairs_by_parallel_sorts(
 }
 // DPCT_LABEL_END
 
+// DPCT_LABEL_BEGIN|segmented_sort_keys_by_parallel_sorts|dpct::internal
+// DPCT_DEPENDENCY_BEGIN
+// DplExtrasAlgorithm|sort_keys
+// DPCT_DEPENDENCY_END
+// DPCT_CODE
+template <typename _ExecutionPolicy, typename key_t, typename key_out_t,
+          typename OffsetIteratorT>
+inline void segmented_sort_keys_by_parallel_sorts(
+    _ExecutionPolicy &&policy, key_t keys_in, key_out_t keys_out, int64_t n,
+    int64_t nsegments, OffsetIteratorT begin_offsets,
+     OffsetIteratorT end_offsets, bool descending = false, int begin_bit = 0,
+    int end_bit = sizeof(typename ::std::iterator_traits<key_t>::value_type) *
+                  8) {
+  using offset_type =
+      typename ::std::iterator_traits<OffsetIteratorT>::value_type;
+  auto host_accessible_offset_starts =
+      sycl::malloc_shared<offset_type>(nsegments, policy.queue());
+  auto host_accessible_offset_ends =
+      sycl::malloc_shared<offset_type>(nsegments, policy.queue());
+  // make offsets accessible on host
+  ::std::copy(::std::forward<_ExecutionPolicy>(policy), begin_offsets,
+              begin_offsets + nsegments, host_accessible_offset_starts);
+  ::std::copy(::std::forward<_ExecutionPolicy>(policy), end_offsets,
+              end_offsets + nsegments, host_accessible_offset_ends);
+
+  for (int i = 0; i < nsegments; i++) {
```

Comment:
Use size_t or uint64_t here? 
- comparison between signed and unsigned integer expressions
- maybe int will overflow.

Reply:
Good catch.  I will fix this.  I think the pair version has the same issue,  If there is no objection, I'll fix that in this same PR.

Reply:
fixed.